libvast/vast/system/file_system.hpp

```diff
@@ -0,0 +1,56 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/fwd.hpp"
+
+#include <caf/fwd.hpp>
+#include <caf/replies_to.hpp>
+#include <caf/typed_actor.hpp>
+
+#include <cstdint>
+
+namespace vast::system {
+
+/// Statistics about filesystem operations.
+struct filesystem_statistics {
+  struct ops_statistics {
+    uint64_t successful = 0;
+    uint64_t failed = 0;
+    uint64_t bytes = 0;
+  };
+
+  ops_statistics writes;
+  ops_statistics reads;
+  ops_statistics mmaps;
+};
+
+// clang-format off
```

Comment:
Is this really necessary here?

Reply:
Yes, the typed actor interface looks not readable otherwise.